Source/Core/DolphinQt/Config/Graphics/HacksWidget.cpp

```diff
@@ -276,6 +279,23 @@ void HacksWidget::AddDescriptions()
       "higher internal resolutions. This setting has no effect when native internal "
       "resolution is used.<br><br><dolphin_emphasis>If unsure, leave this "
       "unchecked.</dolphin_emphasis>");
+  static const char TR_FAST_TEXTURE_SAMPLING_DESCRIPTION[] = QT_TR_NOOP(
+      "Use the video backend's built-in texture sampling functionality instead of a manual "
+      "implementation.<br><br>"
+      "This can introduce graphical issues in some games on some GPUs (such as breaking "
+      "pre-rendered VP6 videos in some EA titles), but will generally (though not always) perform "
+      "better.<br><br>"
+      "Disabling this option uses Manual Texture Sampling, which more accurately handles level of "
+      "detail calculation, and can bypass GPU driver problems.<br><br>"
+      "Manual Texture Sampling also emulates the GameCube/Wii's unusual behavior when wrapping "
+      "textures with non-power-of-2 sizes, but this is disabled when using custom textures or when "
+      "using a higher internal resolution with Scaled EFB Copy enabled.  The accurate texture "
```

Comment:
> The accurate texture wrapping logic is not known to fix any officially released game, but will perform slightly better than the version compatible with custom textures

This is confusing when you previously said:

> This can introduce graphical issues in some games on some GPUs (such as breaking pre-rendered VP6 videos in some EA titles), but will generally (though not always) perform better

Are you just saying that the custom texture code will be less performant than the non-custom texture code for the manual texture sampling?

Reply:
Yes.  The code when custom textures are not in use with manual texture sampling is both faster _and_ more accurate than the code when custom textures are in use with manual texture sampling, since actual hardware uses bitwise and (`&`) to implement texture wrapping, which is faster than modulo (`%`) but behaves weirdly when texture sizes aren't powers of 2.  With custom textures, the assumption that textures will be powers of 2 can be violated, so modulo is used instead, which is slightly slower.

On the other hand, actual game mods might violate the power of 2 size, but it's better for them to get the strange wrapping behavior because that's the behavior they would get on real hardware.  See [issue 12408](https://bugs.dolphin-emu.org/issues/12408) for an example.

Reply:
Modulo arithmetic is also used when Scaled EFB Copy is enabled but shouldn't that only be necessary when the internal resolution is not a power of 2?

Reply:
Theoretically we could get away with using bitwise arithmetic with powers of 2 with scaled EFB copy, yes.  But there's also some rescaling that needs to happen so that texture coordinates are in the right range for the old and new textures, and that can't be skipped.  I think it'd be unnecessarily complicated to special-case the power-of-2 IRs (and special-casing them would also result in more cases where shaders need to be recompiled; currently that only happens when changing between 1x IR and other IRs or when enabling custom textures.)

Reply:
That makes sense.